### PR TITLE
Enable enrich for HttpWebRequest

### LINKED ISSUE #1401: Enrich for HttpClient .NET Framework instrumentation
https://github.com/open-telemetry/opentelemetry-dotnet/pull/1261 introduced `Enrich` API to replace storing raw objects inside Activity. We left out httpclient for .NET framework from that PR. Opening an issue to track it.

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #1401

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs [hunk 1]
```diff
@@ -104,7 +98,15 @@ private static void AddRequestTagsAndInstrumentRequest(HttpWebRequest request, A
 
             if (activity.IsAllDataRequested)
             {
-                activity.SetCustomProperty(RequestCustomPropertyName, request);
+                try
+                {
+                    Options.Enrich?.Invoke(activity, "AddRequestTagsAndInstrumentRequest", request);
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:103]: We can name it ActivityStart

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs [hunk 2]
```diff
@@ -120,7 +122,15 @@ private static void AddResponseTags(HttpWebResponse response, Activity activity)
         {
             if (activity.IsAllDataRequested)
             {
-                activity.SetCustomProperty(ResponseCustomPropertyName, response);
+                try
+                {
+                    Options.Enrich?.Invoke(activity, "AddResponseTags", response);
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:127]: call it StopActivity

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs [hunk 3]
```diff
@@ -138,7 +148,14 @@ private static void AddExceptionTags(Exception exception, Activity activity)
                 return;
             }
 
-            activity.SetCustomProperty(ExceptionCustomPropertyName, exception);
+            try
+            {
+                Options.Enrich?.Invoke(activity, "AddExceptionTags", exception);
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:153]: OnException

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs (hunk 1 above)
Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:103]: should we rename method and string to OnStart, OnStop, OnException just like the others?

File: src/OpenTelemetry.Instrumentation.Http/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+* Instrumentation no longer store raw objects like `HttpWebRequest` in
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Http/CHANGELOG.md:5]: Lets be more explicit that this PR is fixing this for HttpClient in .NET Framework. HttpClient in .NET Core already had Enrich

Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Http/CHANGELOG.md:5]: can you modify the README.md as well to reflect this change?

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:103]: method is fine. OnStartActivity, OnStopActivity, OnException are the names used in .net core, so lets keep it same here.

Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:103]: updated!

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs (hunk 2 above)
Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs]: updated!

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs (hunk 3 above)
Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs]: updated!

File: src/OpenTelemetry.Instrumentation.Http/CHANGELOG.md (hunk 4 above)
Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Http/CHANGELOG.md]: Added the following text: "Instrumentation for `HttpWebRequest`...."

**Review by cijothomas — APPROVED** [review]

**Pull request merged**